packages/bitcore-payid/src/index.ts

```diff
@@ -1,24 +1,42 @@
 import {
   AddressDetailsType,
+  getDefaultAlgorithm,
+  IdentityKeySigningParams,
+  signWithKeys,
+  toKey,
   verifySignedAddress,
 } from '@payid-org/utils';
-import { GeneralJWS, IVerifyPayId } from './index.d';
+import Bitcore from 'bitcore-lib';
+import * as errors from './errors';
+import { GeneralJWS, IVerifyPayId, JWK } from './index.d';
 
 class PayId {
   constructor() {}
 
   /**
    * Sign a payId address with the given identityKey
-   * @param {string} payId e.g.: "alice.smith$bitpay.com", "bob.acosta$example.com"
+   * @param {string} payId e.g.: "alice.smith$bitpay.com", "bob123$example.com"
    * @param {string} address BTC, ETH, or XRP address to be signed
    * @param {string} currency Currency ticker (e.g. "BTC", "ETH", "XRP")
-   * @param {string} identityKey The key to be used for signing.
-   *      Must be an ASN.1 formattable key. https://en.wikipedia.org/wiki/ASN.1
-   *      Can be a asynchronous (RSA, EC) private key or a synchronous (Oct, OTP) key and can have any length.
+   * @param {string} identityKey The private key to be used for signing.
+   *      Must be an asynchronous (RSA, EC) key and can have any length.
+   *      **If not from bitcore-lib and is a string, it must be a PEM string**
    */
-  sign(payId: string, address: string, currency: string, identityKey: string): GeneralJWS {
-    // TODO
-    return {} as GeneralJWS;
+  sign(payId: string, address: string, currency: string, identityKey: string | Buffer): GeneralJWS {
+    let jwk = this._convertIdentityKeyToJWK(identityKey);
+
+    const signingParams = new IdentityKeySigningParams(jwk, getDefaultAlgorithm(jwk));
+    const addy = {
+      paymentNetwork: currency,
+      environment: 'mainnet',
```

Comment:
maybe add an optional param 'environment' which defaults to `mainnet`?  so we have the option in the future.

Reply:
Done